etcdserver/server.go

```diff
@@ -745,9 +747,19 @@ func (s *EtcdServer) apply(es []raftpb.Entry, confState *raftpb.ConfState) (uint
 				}
 				break
 			}
-			var r pb.Request
-			pbutil.MustUnmarshal(&r, e.Data)
-			s.w.Trigger(r.ID, s.applyRequest(r))
+
+			var raftReq pb.InternalRaftRequest
+			if !pbutil.MaybeUnmarshal(&raftReq, e.Data) { // backward compatible
```

Comment:
Ack. If we add requestv3 into InternalRaftRequest, we should be careful that its wire type is different from related tag in Request.

Maybe we could add a conflict wire type field into InternalRaftRequest. But it is totally ugly.